neutrons/Shiver
# OnCat widget

## Description

This PR introduces the following changes:

- New OnCat widget is added to allow Shiver to retrieve metadata from OnCat.
- Corresponding unit tests.
- Cross widget connection between data selection and OnCat widget.

## Test instructions

- Clone and setup the feature branch as instructed in the description.
- Start the application and go to the generate page
  ![image](https://user-images.githubusercontent.com/5064852/236266954-aec20c63-1bca-4cda-a741-952adac22074.png)
  - first time, the connection status should be in red.  Once logged in, a token will be saved and shiver can used to automatically connect to OnCat within 24h.
- click `connect to Oncat` and type in UCAM
  ![image](https://user-images.githubusercontent.com/5064852/236267057-b84a33a2-1387-4ca5-a99e-aa97edf69244.png)
- now you should be able to navigate between different instrument, ipts and dataset from the oncat widget. If on analysis cluster, the data widget will also be auto updated if files can be found.
  ![image](https://user-images.githubusercontent.com/5064852/236267266-fd0586e1-6e02-4cb6-ab67-198d455f582a.png)

## Known issues
The unit test  `tests/views/test_sample_parameters_buttons.py` leads to fatal Python error when testing locally and is marked as skipped.
The corresponding error messages are added to the docstring of the skipped test.


> IBM item: [Story939](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=939)
